File: scalaz/src/main/scala/org/openapitools/client/api/MailAttachment.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.io.File

import MailAttachment._

case class MailAttachment (
  /* Optional filename to specify for the attachment. */
  filename: Option[String],
/* Contents of the attached file */
  data: File)

object MailAttachment {
  import DateTimeCodecs._

  implicit val MailAttachmentCodecJson: CodecJson[MailAttachment] = CodecJson.derive[MailAttachment]
  implicit val MailAttachmentDecoder: EntityDecoder[MailAttachment] = jsonOf[MailAttachment]
  implicit val MailAttachmentEncoder: EntityEncoder[MailAttachment] = jsonEncoderOf[MailAttachment]
}

File: openapi/javascript-closure-angular/API/Client/MailAttachment.js
goog.provide('API.Client.MailAttachment');

/**
 * A File attachment for an email
 * @record
 */
API.Client.MailAttachment = function() {}

/**
 * Optional filename to specify for the attachment.
 * @type {!string}
 * @export
 */
API.Client.MailAttachment.prototype.filename;

/**
 * Contents of the attached file
 * @type {!Object}
 * @export
 */
API.Client.MailAttachment.prototype.data;

